buildSrc/src/main/groovy/modernfix.mod-common-conventions.gradle

```diff
@@ -16,37 +16,26 @@ dependencies {
             parchment("org.parchmentmc.data:parchment-${minecraft_version}:${parchment_version}@zip")
         }
     }
+    implementation project(":annotations")
+    annotationProcessor project(path: ":annotation-processor", configuration: 'shadow')
 }
 
+project.sourceSets {
+    main.resources.srcDirs += [layout.buildDirectory.dir("generated/sources/annotationProcessor/java/main/resources")]
+}
+// hack to shut up gradle about the hack to include generated resources
 tasks {
     processResources {
-        def mixinFileList = []
-        def mixinDirectory = file("src/main/java/org/embeddedt/modernfix/" + project.name + "/mixin")
-        fileTree(mixinDirectory).visit { FileVisitDetails details ->
-            if(details.file.isFile()) {
-                def fileName = mixinDirectory.relativePath(details.file).toString().replaceFirst(/\.java$/, "").replace('/', '.')
-                mixinFileList << fileName
-            }
-        }
-
-        def mixinClassesStringB = new StringBuilder()
-        for(int i = 0; i < mixinFileList.size(); i++) {
-            mixinClassesStringB.append("    \"")
-            mixinClassesStringB.append(mixinFileList.get(i))
-            mixinClassesStringB.append('"')
-            if(i < (mixinFileList.size() - 1))
-                mixinClassesStringB.append(',')
-            mixinClassesStringB.append('\n')
-        }
-
-        def replacements = [
-                mixin_classes: mixinClassesStringB.toString()
-        ]
+        dependsOn compileJava
+    }
+}
 
-        inputs.properties replacements
-        def filePattern = "modernfix-" + project.name + ".mixins.json"
-        filesMatching(filePattern) {
-            expand replacements
+tasks.withType(JavaCompile) {
+    options.fork = true
+    options.forkOptions.jvmArgs << '--enable-preview'
```

Comment:
Why does the mod itself need to be compiled with `--enable-preview`? It conflicts with the commit description, which says it's only needed for the AP: https://github.com/embeddedt/ModernFix/commit/4645c2622742d62ecf9fd08028897edb40512be3

Reply:
So the AP itself is compiled with preview... but you need to enable preview features on the JVM to run the AP. This hunk specifically enables it for the compilation process/AP but not the code being compiled.
See https://github.com/gradle/gradle/issues/12680